src/backend/mi2/mi2.ts

```diff
@@ -591,12 +654,17 @@ export class MI2 extends EventEmitter implements IBackend {
 				if (breakpoint.countCondition[0] == ">")
 					location += "-i " + numRegex.exec(breakpoint.countCondition.substring(1))[0] + " ";
 				else {
+					if (!isPositiveInteger(breakpoint.countCondition)) {
+						this.log("stderr", "Unsupported break count expression: '" + breakpoint.countCondition + "'. Only supports 'X' for breaking once after X times or '>X' for ignoring the first X breaks");
+						resolve([false, undefined]);
+						return;
+					}
 					const match = numRegex.exec(breakpoint.countCondition)[0];
 					if (match.length != breakpoint.countCondition.length) {
 						this.log("stderr", "Unsupported break count expression: '" + breakpoint.countCondition + "'. Only supports 'X' for breaking once after X times or '>X' for ignoring the first X breaks");
 						location += "-t ";
 					} else if (parseInt(match) != 0)
-						location += "-t -i " + parseInt(match) + " ";
+						location += "-t -i " + (parseInt(match) - 1) + " ";
```

Comment:
change in behavior, revert or put into separate PR with changelog if there is a reason for this